### PR TITLE
Jgfouca/upstream merge 2024 02 21

### PR DESCRIPTION
**Pull request by jgfouca**
No real conflicts but there were a couple issues, both related to YAKL...

1. YAKL apparently changed its cmake to define itself as an INTERFACE library, which forced us to change a few things in our CMake.
2. Mappy was hitting YAKL out-of-memory errors, so I had to GATOR_DISABLE

### CODE REVIEW

File: .github/workflows/e3sm-gh-ci-cime-tests.yml
[the remark this answers is not in the document]
Comment (jgfouca) [reply .github/workflows/e3sm-gh-ci-cime-tests.yml:1]: I think it's fine as long as it's pretty quick to run.

**Review by bartgol — DISMISSED** [review]

**Comment by ambrad** [comment]
Jim, what's the specific code you modified associated with the GATOR_DISABLE setting?

**Comment by jgfouca** [comment]
@ambrad , I modified the config_machines.xml entry for mappy to add GATOR_DISABLE=1 to the CIME env.

**Comment by jgfouca** [comment]
I think this can probably be merged, assuming weaver passed on the next round of AT. Mappy is showing DIFFs for v0 cases but not v1. I have no idea how concerning that is; my guess is we can probably just bless these. There were yet more problems with YAKL (not YAKL in itself but our integration with it).

**Review by bartgol — APPROVED** [review]

**Pull request merged**